pkg/proxy/conntrack/conntrack.go

```diff
@@ -69,7 +75,13 @@ func (ct *conntracker) ClearEntries(ipFamily uint8, filters ...netlink.CustomCon
 		return 0, nil
 	}
 
-	n, err := ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, netlink.InetFamily(ipFamily), filters...)
+	var n uint
+	var err error
+	err = retry.OnError(util.MaxAttemptsEINTR, util.ShouldRetryOnEINTR, func() error {
+		n, err = ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, netlink.InetFamily(ipFamily), filters...)
+		return err
+	})
```

Comment:
🤔 
I'm unsure if this is correct. Since this is deleting entries, I assume the partial result returned is a list of deleted entries.
Retrying here until an error isn't returned means that the logs that are created below this retry, and over at https://github.com/kubernetes/kubernetes/blob/b828fc7701b5b25974fcb230f5bdd59318be58bf/pkg/proxy/conntrack/cleanup.go#L123-L127 could be incorrect

Should the retry logic be moved elsewhere? Or something else be changed here?

Reply:
My worry is a scenario like this...
Assume we have 100 entries that need deleting.
1. On first try, 20 are deleted, and the partial result error is returned
2. We retry
3. On second try, the remaining 80 are deletes, and no error is returned
4. The number of entries (80) is passed back to the caller, and the log says `"entriesDeleted", 80`, which is incorrect, since a total of 100 were deleted

I guess either the tally of records can be summed during the retry, or the retry logic can move.